libs/community/langchain_community/chat_models/mlflow.py

```diff
@@ -119,9 +126,72 @@ def _generate(
             data["stop"] = stop
         if self.max_tokens is not None:
             data["max_tokens"] = self.max_tokens
+
+        return data
+
+    def _generate(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> ChatResult:
+        data = self._prepare_inputs(
+            messages, stop, **kwargs,
+        )
         resp = self._client.predict(endpoint=self.endpoint, inputs=data)
         return ChatMlflow._create_chat_result(resp)
 
+    def stream(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        *,
+        stop: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> Iterator[BaseMessageChunk]:
+        # We need to override `stream` to handle the case
+        # that `self._client` does not implement `predict_stream`
+        if not hasattr(self._client, "predict_stream"):
+            # MLflow deployment client does not implement streaming,
+            # so use default implementation
+            yield cast(
+                BaseMessageChunk, self.invoke(input, config=config, stop=stop, **kwargs)
+            )
```

Comment:
why not do this check in _stream?

Reply:
`checking it in `_stream` does not work.

When `chat_databricks_model.stream()` is invoked, if `ChatDatabricks` class does not override `stream` method but only override `_stream`, then this [line](https://github.com/WeichenXu123/langchain/blob/3218463f6a3d841905648971735949a14a16e191/libs/core/langchain_core/language_models/chat_models.py#L193) is executed, 

then `if type(self)._stream == BaseChatModel._stream` is always False, then it goes into this [branch](https://github.com/WeichenXu123/langchain/blob/3218463f6a3d841905648971735949a14a16e191/libs/core/langchain_core/language_models/chat_models.py#L199), i.e. it doesn't invoke `_stream`

Reply:
> why not do this check in _stream?

If so, before executing
```
yield cast(
                BaseMessageChunk, self.invoke(input, config=config, stop=stop, **kwargs)
            )
```,

the `BaseChatModel.stream` will execute these lines code:

```
            config = ensure_config(config)
            messages = self._convert_input(input).to_messages()
            params = self._get_invocation_params(stop=stop, **kwargs)
            options = {"stop": stop, **kwargs}
            callback_manager = CallbackManager.configure(
                config.get("callbacks"),
                self.callbacks,
                self.verbose,
                config.get("tags"),
                self.tags,
                config.get("metadata"),
                self.metadata,
            )
            (run_manager,) = callback_manager.on_chat_model_start(
                dumpd(self),
                [messages],
                invocation_params=params,
                options=options,
                name=config.get("run_name"),
                run_id=config.pop("run_id", None),
                batch_size=1,
            )
```
this causes different behavior , 

I think if stream is not supported, `model.stream()` should only trigger 

```
            yield cast(
                BaseMessageChunk, self.invoke(input, config=config, stop=stop, **kwargs)
            )
```
to keep the consistent behavir with master branch.